src/rezplugins/shell/_utils/powershell_base.py

```diff
@@ -23,6 +23,15 @@ class PowerShellBase(Shell):
     expand_env_vars = True
     syspaths = None
 
+    # PowerShell does not provide built-in functionality to manually trigger
+    # sourcing of the user/host profile scripts in the appropriate order, so we
+    # have to implement it ourselves in order to be able to control the relative
+    # source order of the user's profile vs. the generated rez context script.
+    profile_source_cmd = (
+        '$PROFILE '
+        '| Get-Member -type NoteProperty '
+        '| ForEach-Object { if (Test-Path $PROFILE."$($_.Name)") { . $PROFILE."$($_.Name)" }}')
```

Comment:
Note that this is a dynamic approach to discovering available profile scripts, but it may be preferable to hardcode iteration over the known profile slots [documented here](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_profiles?view=powershell-7.4#profile-types-and-locations).

This could look something like this (+/- formatting):
```pwsh
foreach ($name in 'AllUsersAllHosts', 'AllUsersCurrentHost', 'CurrentUserAllHosts', 'CurrentUserCurrentHost')
{
    if (Test-Path $PROFILE.$name) {
        . $PROFILE.$name
    }
}
```